executable-spec/src/Cardano/Ledger/Spec/STS/Update/Ideation.hs

```diff
@@ -92,26 +107,61 @@ instance HasTrace IDEATION where
   sigGen
     _maybePredicateFailure
     participants
-    Data.State { submittedSIPs, revealedSIPs }
-    =
-    Gen.frequency [ (1, generateASubmission)
-                  , (1, generateARevelation)
-                  ]
-    where
-      -- Generate a submission taking a participant that hasn't submitted a proposal yet
-      generateASubmission
-        = fmap (Submit . (nextId `Data.SIP`))
-        $ Gen.element
-        $ Set.toList
-        $ dom participants
-        where
-          nextId = maximum $ 0 : fmap (1+) ids
-          ids =  Set.toList (Set.map Data.id submittedSIPs)
-              ++ Set.toList (Set.map Data.id revealedSIPs)
-
-      generateARevelation =
-        case Set.toList submittedSIPs of
-          [] -> mzero -- We cannot generate a revelation, since there are no
-                      -- (SIP) commitments yet, so we retry (in this case
-                      -- eventually we'll generate a submission).
-          xs -> fmap Reveal $ Gen.element xs
+    Data.State { submittedSIPs, revealedSIPs } = do
+      owner <- newOwner
+      Gen.frequency [ (1, generateASubmission owner)
+                    , (1, generateARevelation)
+                    ]
+
+      where
+        !newOwner = -- (fmap (Data.author) newSIP)
```

Comment:
Why the exclamation mark here?

Reply:
It was left from a debugging effort :-)
I will remove it.

Reply:
I have removed it.